src/pager/index.js

```diff
@@ -0,0 +1,118 @@
+     * The number of items that are displayed in each page.
+     */
+    perPage: PropTypes.number,
+    /**
+     * The total number of items that are available.
+     */
+    totalItemCount: PropTypes.number.isRequired,
+    /**
+     * This callback is called when the user navigates to a different page. It
+     * has a single argument which is the new page number.
+     */
+    onPageChangeCallback: PropTypes.func.isRequired,
+    /**
+     * The classname of the page element that is currenting displayed.
+     */
+    selectedClassName: PropTypes.string
+  };
+
+  static defaultProps = {
+    perPage: 10,
+    selectedClassName: ''
+  };
+
+  render() {
+    let {currentPage} = this.props;
+    return (
+      <div className='pt-pager' {...this.props}>
+        <span className='pt-next-page' onClick={this._handleClick.bind(null, currentPage-1)}>
+          <Icon icon='chevron-left'/>
+        </span>
+        {this._getPageNumbers()}
+        <span className='pt-previous-page'  onClick={this._handleClick.bind(null, currentPage+1)}>
+          <Icon icon='chevron-right'/>
+        </span>
+      </div>
+    );
+  }
+
+  _getPageNumbers = () => {
+    let { currentPage } = this.props;
+    let currentPageIndex = currentPage - 1;
+    let totalPageCount = this._getTotalPageCount();
+
+    let pages = Array.from(Array(totalPageCount).keys());
+    // The simulated current page is required to make sure that we always show at least 5 numbers around
+    // the current page (even if current page is in the extremes)
+    let simulatedCurrentPageIndex = currentPageIndex;
+    if (currentPageIndex < EDGE_WINDOW_SIZE) {
+      simulatedCurrentPageIndex = EDGE_WINDOW_SIZE;
+    }
+    else if (currentPageIndex > totalPageCount - EDGE_WINDOW_SIZE - 1) {
+      simulatedCurrentPageIndex = totalPageCount - EDGE_WINDOW_SIZE - 1;
+    }
+
+    let firstEllipsisStartIndex = EDGE_WINDOW_SIZE;
+    let firstEllipsisCount = (simulatedCurrentPageIndex - Math.floor(MOVING_WINDOW_SIZE/2)) - firstEllipsisStartIndex;
+
+    let secondEllipsisStartIndex = simulatedCurrentPageIndex + Math.ceil(MOVING_WINDOW_SIZE/2);
+    let secondEllipsisCount = (totalPageCount - EDGE_WINDOW_SIZE) - secondEllipsisStartIndex
+
+    // Remove and replace any pages that will be ellipsed with -1
+    if (secondEllipsisCount > 0) {
+      pages.splice(secondEllipsisStartIndex, secondEllipsisCount, -1);
+    }
+    if (firstEllipsisCount > 0) {
+      pages.splice(firstEllipsisStartIndex, firstEllipsisCount, -1).length;
+    }
+
+    return pages.map((pageIndex) => {
+      // If it's -1, it's an ellipsised section.
+      if (pageIndex == -1) {
+        return this._getEllipsis();
+      }
+      else {
+        return this._getSpanForPage(pageIndex+1);
+      }
+    })
+  }
+
+  _getSpanForPage = (pageNum) => {
+    let {currentPage, selectedClassName} = this.props;
+    let classes = 'pt-page-element';
+    if (currentPage == pageNum) {
+      classes += ' ' + selectedClassName;
```

Comment:
I wish the syntax for dynamic classnames used in classnames library were not so ugly. As it stands it would look like this: 

``` 
classnames('pt-page-element', { [`${selectedClassName}`]: currentPage == pageNum })
```